libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -279,6 +279,7 @@ data Binder fn a where
 deriving instance Show (Binder fn a)
 
 data Pred (fn :: [Type] -> Type -> Type) where
+  Monitor :: ((forall a. Term fn a -> a) -> Property -> Property) -> Pred fn
```

Comment:
Maybe an example of how this is intended to be used?

Reply:
I gave `monitor` and `monitorSpec` some explanatory haddocks.